src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```diff
@@ -15,11 +15,16 @@
 
     <RuntimeIlcPackageName>runtime.$(_targetOS)-$(_targetArchitecture).Microsoft.DotNet.ILCompiler</RuntimeIlcPackageName>
 
+    <IlcHostOS>linux</IlcHostOS>
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">osx</IlcHostOS>
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('FREEBSD'))">freebsd</IlcHostOS>
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">windows</IlcHostOS>
+
     <!-- Default to host that matches OS architecture on Windows. It allows compilation using msbuild.exe that is x86 by default -->
     <IlcHostArch Condition="'$(OS)' == 'Windows_NT' and '$(IlcHostArch)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant)</IlcHostArch>
     <!-- Default to host that matches SDK architecture on non-Windows -->
     <IlcHostArch Condition="'$(IlcHostArch)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</IlcHostArch>
-    <IlcHostPackageName>runtime.$(_targetOSPkg)-$(IlcHostArch).Microsoft.DotNet.ILCompiler</IlcHostPackageName>
+    <IlcHostPackageName>runtime.$(IlcHostOS)-$(IlcHostArch).Microsoft.DotNet.ILCompiler</IlcHostPackageName>
```

Comment:
This was the only use of `_targetOSPkg` property. We can delete the `_targetOSPkg` property - revert https://github.com/dotnet/runtime/pull/82590/files.

Reply:
Good catch. I rebased it and didn't notice.

Reply:
Reverting will bring back the linux-musl bug? We don’t have CI coverage for these, so I test it once the next daily build of installer is available.